synthesizer/src/vm/verify.rs

```diff
@@ -91,7 +91,7 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
 
         // Compute the Merkle root of the transaction.
         // Debug-mode only, as the `Transaction` constructor recomputes the transaction ID at initialization.
-        #[cfg(debug_assertions)]
+        // Attention - This check is mandatory. This is the only way to ensure that the transaction ID is well-formed.
```

Comment:
Earlier reviews had concerns with keeping and removing this check.

**For keeping.** Transaction contents can be modified, which would result in an incorrect transaction ID.

**Against keeping.** The constructors `from_deployment` and `from_execution` recompute the transaction ID at initialization, which involves computing the root of the transaction tree. We know from manual code review that validators use those constructors. Removing this check was needed to scale up the validator set.

Reply:
On looking through the code again, I could not find any constructors that did not recompute the root. If that's the case, then an ID for a `Transaction` in memory is correct-by-construction for an honest validator. 

@reviewers do you agree?

Tangent. I noticed that `get_transaction` on the Ledger performs this check, making it a pretty expensive operation to do, especially for our REST API.

Reply:
Well for the public record, I agree, but you knew :)